src/Moq/Mock.cs

```diff
@@ -508,42 +508,17 @@ private static VoidSetupPhrase<T> SetupPexProtected<T>(Mock<T> mock, Expression<
 		}
 
 		[DebuggerStepThrough]
-		internal static SetterSetupPhrase<T, TProperty> SetupSet<T, TProperty>(
-			Mock<T> mock,
-			Action<T> setterExpression,
-			Condition condition)
-			where T : class
+		internal static MethodCall SetupSet(Mock mock, Delegate setterExpression, Condition condition)
 		{
-			return PexProtector.Invoke(SetupSetPexProtected<T, TProperty>, mock, setterExpression, condition);
+			return PexProtector.Invoke(SetupSetPexProtected, mock, setterExpression, condition);
 		}
 
-		private static SetterSetupPhrase<T, TProperty> SetupSetPexProtected<T, TProperty>(
-			Mock<T> mock,
-			Action<T> setterExpression,
-			Condition condition)
-			where T : class
+		private static MethodCall SetupSetPexProtected(Mock mock, Delegate setterExpression, Condition condition)
 		{
 			var (m, expr, method, value) = SetupSetImpl(mock, setterExpression);
-			Debug.Assert(value.Length == 1);
```

Comment:
If we'd had a unit test calling this method for an indexer set expression, we'd have seen this assertion fail. Let's remove it.